core/src/main/java/org/apache/spark/shuffle/external/ExternalShuffleDataIO.java

```diff
@@ -0,0 +1,54 @@
+package org.apache.spark.shuffle.external;
+
+import org.apache.spark.SparkConf;
+import org.apache.spark.SparkEnv;
+import org.apache.spark.network.TransportContext;
+import org.apache.spark.network.netty.SparkTransportConf;
+import org.apache.spark.network.server.NoOpRpcHandler;
+import org.apache.spark.network.util.TransportConf;
+import org.apache.spark.shuffle.api.ShuffleDataIO;
+import org.apache.spark.shuffle.api.ShuffleReadSupport;
+import org.apache.spark.shuffle.api.ShuffleWriteSupport;
+import org.apache.spark.SecurityManager;
+import org.apache.spark.storage.BlockManager;
+
+public class ExternalShuffleDataIO implements ShuffleDataIO {
+
+    private final TransportConf conf;
+    private final TransportContext context;
+    private static BlockManager blockManager;
+    private static SecurityManager securityManager;
+    private static String hostname;
+    private static int port;
+
+    public ExternalShuffleDataIO(
+            SparkConf sparkConf) {
+        this.conf = SparkTransportConf.fromSparkConf(sparkConf, "shuffle", 1);
+        // Close idle connections
+        this.context = new TransportContext(conf, new NoOpRpcHandler(), true, true);
+    }
+
+    @Override
+    public void initialize() {
```

Comment:
I started playing with this code a bit, and pretty quickly run into this: it would be good for the interfaces to declare at least `throws IOException`, otherwise implementations will have to write extra code for no good reason.